src/storage-types/src/sources/load_generator.rs

```diff
@@ -447,8 +472,68 @@ impl RustType<ProtoLoadGeneratorSourceConnection> for LoadGeneratorSourceConnect
                     count_clerk,
                 },
                 Kind::Datums(()) => LoadGenerator::Datums,
+                Kind::KeyValue(kv) => LoadGenerator::KeyValue(kv.into_rust()?),
             },
             tick_micros: proto.tick_micros,
         })
     }
 }
+
+#[derive(Arbitrary, Clone, Debug, Eq, PartialEq, Serialize, Deserialize, Default)]
+pub struct KeyValueLoadGenerator {
+    // The keyspace of the source.
+    pub keys: u64,
+    // The number of rounds to emit values for each key in the snapshot.
+    // This lets users scale the snapshot size independent of the keyspace.
+    pub snapshot_rounds: u64,
+    // The number of rounds to emit values for each key, after the snapshot, before
+    // respecting the `update_rate`.
+    //
+    // This lets us quickly produce updates, as opposed to a single-value per key during
+    // snapshotting.
+    pub quick_rounds: u64,
+    // The number of random bytes for each value.
+    pub value_size: u64,
+    // The number of partitions. The keyspace is spread evenly across the partitions.
+    // This lets users scale the concurrency of the source independently of the replica size.
+    pub partitions: u64,
+    // If provided, the maximum rate at which new batches of updates, per-partition will be
+    // produced after the snapshot.
+    pub update_rate: Option<Duration>,
+    // The number of keys in each update batch.
+    pub batch_size: u64,
+    // A per-source seed.
+    pub seed: u64,
+    // Whether or not to include the offset in the value. The string is the column name.
+    pub include_offset: Option<String>,
+}
+
+impl RustType<ProtoKeyValueLoadGenerator> for KeyValueLoadGenerator {
+    fn into_proto(&self) -> ProtoKeyValueLoadGenerator {
+        ProtoKeyValueLoadGenerator {
+            keys: self.keys,
+            snapshot_rounds: self.snapshot_rounds,
+            quick_rounds: self.quick_rounds,
+            value_size: self.value_size,
+            partitions: self.partitions,
+            update_rate: self.update_rate.as_ref().map(|ur| ur.into_proto()),
```

Comment:
We have generic implementations for this

```suggestion
            update_rate: self.update_rate.into_proto(),
```

Reply:
oh nice!